### ISSUE TITLE
Charade dataloader - Loading testing data

### ISSUE THREAD

**Issue by adroit89** [dairui01/MS-TCT#10]
Hello, I am looking at the Charades data loader and it seems like when loading the features for a video from the test set, the current code only loads the first "num_clips" features as "random_index" is set to 0.

https://github.com/dairui01/MS-TCT/blob/7e5e8793177eadb036688353fce11d4d85fe5fa6/charades_dataloader.py#L100

How about the remainder of the video? Is that not used for evaluation? Please correct me if I am wrong here.

**Comment by dairui01** [comment]
Charades dataset contains only short-term video (about 30 sec.), thus at inference, with `random_index=0` and the chosen number of clips is enough. You can also change the `num_clips` to 256 or 512, the performance should not drop by this hyperparameter.

**Comment by adroit89** [comment]
How do you handle this for MultiTHUMOS? Do you do a sliding window during inference?

**Comment by dairui01** [comment]
Yes. For long-term video as MultiTHUMOS (2-3 mins), we follow the same training strategy but use a sliding window approach at inference.
